### PR TITLE
Initial support for snapshot deployment

### PR DESCRIPTION
**Pull request by cescoffier**
Just a first version.
Issue-based tracking will be done once this is working.

Unfortunately, `deployAtEnd` does not work because our tests influence the number of projects.

### CODE REVIEW

File: .github/workflows/ci-deploy-snapshots.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Build-And-Deploy
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_API_TOKEN }}
+          SERVER_USERNAME: ${{ secrets.SERVER_USERNAME }}
+          SERVER_PASSWORD: ${{ secrets.SERVER_PASSWORD }}
+        run: |
+          mvn -e -B --settings .github/mvn-settings.xml \
+            -DskipITs -Dno-format -Dinvoker.skip=true \
+            -DretryFailedDeploymentCount=10 \
+            clean deploy
```
Comment (gastaldi) [inline .github/workflows/ci-deploy-snapshots.yml:42]: I think you also want to `-DskipTests`?

Comment (gastaldi) [reply .github/workflows/ci-deploy-snapshots.yml:42]: Or `-Dmaven.test.skip` to skip test compilation also

Comment (cescoffier) [reply .github/workflows/ci-deploy-snapshots.yml:42]: @gsmet wanted to run some tests.

**Review by gsmet — APPROVED** [review]
@cescoffier I pushed a second commit with some minor adjustments. The most important one is that it's better to build with JDK 8 as long as we support it.

Feel free to squash if it's good for you.

**Comment by cescoffier** [comment]
squashed and merged.

**Pull request merged**